packages/flutter_tools/lib/src/android/deferred_components.dart

```diff
@@ -0,0 +1,674 @@
+      if (!androidFiles.verifyFilesExist()) {
+        // generate into temp directory
+        generatedFiles.addAll(
+          await androidFiles.generateFiles(
+            alternateAndroidDir: _outputDir,
+            clearAlternateOutputDir: true,
+          )
+        );
+      }
+    }
+    return generatedFiles;
+  }
+
+  List<String> _checkAndroidStringMapping(List<DeferredComponent> components, List<LoadingUnit> generatedLoadingUnits) {
+    Directory androidDir = env.projectDir.childDirectory('android');
+    List<String> modifiedFiles = <String>[];
+    List<String> lines;
+    // TODO(garyq): Re-work this to be much more robust and use Xml lib.
+    // Add mapping to strings.xml
+    File stringRes = androidDir.childDirectory('app').childDirectory('src').childDirectory('main').childDirectory('res').childDirectory('values').childFile('strings.xml');
+    File stringResOutput = _outputDir.childDirectory('app').childDirectory('src').childDirectory('main').childDirectory('res').childDirectory('values').childFile('strings.xml');
+    bool needsStrings = true;
+    lines = stringRes.readAsLinesSync();
+    List<String> requiredLines  = <String>[];
+    for (DeferredComponent component in components) {
+      requiredLines.add('<string name="${component.name}Name">${component.name}</string>');
+      Set<int> usedIds = <int>{};
+      for (LoadingUnit loadingUnit in generatedLoadingUnits) {
+        if (usedIds.contains(loadingUnit.id)) {
+          continue;
+        }
+        for (String lib in component.libraries) {
+          if (loadingUnit.libraries.contains(lib)) {
+            requiredLines.add('<string name="loadingUnit${loadingUnit.id}">${component.name}</string>');
+            usedIds.add(loadingUnit.id);
+            break;
+          }
+        }
+      }
+    }
+    if (stringRes.existsSync()) {
+      Set<String> remainingLines = <String>{};
+      remainingLines.addAll(requiredLines);
+      for (String line in lines) {
+        for (String pattern in remainingLines) {
+          if (line.contains(pattern)) {
+            remainingLines.remove(pattern);
+            break;
+          }
+        }
+      }
+      if (remainingLines.isNotEmpty) {
+        for (String line in lines) {
+          if (line.contains('<resources>')) {
+            stringResOutput.writeAsStringSync('$line\n', mode: FileMode.append, flush: true);
+            for (String value in remainingLines) {
+              stringResOutput.writeAsStringSync('    $line\n', flush: false, mode: FileMode.append);
+            }
+            continue;
+          }
+          stringResOutput.writeAsStringSync('$line\n', mode: FileMode.append, flush: true);
+        }
+        modifiedFiles.add(stringResOutput.path);
+      }
+    } else { // Generate completely new file
+      stringResOutput.createSync(recursive: true);
+      stringResOutput.writeAsStringSync(
+'''
```

Comment:
didn't you make a file for this?

Reply:
Not quite. This is strings.xml. The contents here are too dynamic for a template to work.

Reply:
I want to try to change this to use AndroidManifest.xml + applicationInfo to store the mapping rather than strings